packages/build/src/steps/run_core_step.js

```diff
@@ -0,0 +1,209 @@
+import { getBufferLogs } from '../log/logger.js'
+import { logBuildStart } from '../log/messages/core.js'
+import { bundleEdgeFunctions } from '../plugins_core/edge_functions/index.js'
+import { reportStatuses } from '../status/report.js'
+
+import { runSteps } from './run_steps.js'
+
+/**
+ * Executes specified build steps for functions
+ *
+ * @param  {object} [flags] - build configuration CLI flags
+ * @param  {string} [flags.config] - Path to the configuration file
+ * @param  {string} [flags.cwd] - Current directory. Used to retrieve the configuration file
+ * @param  {string} [flags.repositoryRoot] - Git repository root directory. Used to retrieve the configuration file.
+ * @param  {string} [flags.apiHost] - Netlify API endpoint
+ * @param  {string} [flags.token] - Netlify API token for authentication
+ * @param  {string} [flags.siteId] - Netlify Site ID
+ * @param  {string} [flags.deployId] - Netlify Deploy ID
+ * @param  {string} [flags.context] - Build context
+ * @param  {string} [flags.branch] - Repository branch
+ * @param  {boolean} [flags.dry=false] - Run in dry mode, i.e. printing steps without executing them
+ * @param  {string} [flags.nodePath] - Path to the Node.js binary to use in the build command and plugins
+ * @param  {boolean} [flags.buffer=false] - Buffer output instead of printing it
+ *
+ * @returns {object} buildResult
+ * @returns {boolean} buildResult.success - Whether build succeeded or failed
+ * @returns {number} buildResult.severityCode - Build success/failure status among:
+ * 0 (success), 1 (build cancelled), 2 (user error), 3 (plugin error), 4 (system error). Can be used as exit code.
+ * @returns {string[]} buildResult.logs - When using the `buffer` option, all log messages
+ */
+export default async function runCoreStep(buildSteps, flags = {}) {
+  const { errorMonitor, mode, logs, debug, ...flagsA } = startBuild(flags)
+
+  const errorParams = { errorMonitor, mode, logs, debug }
+
+  try {
+    const { netlifyConfig: netlifyConfigA, configMutations } = await executeBuildStep({
+      ...flagsA,
+      errorMonitor,
+      mode,
+      logs,
+      debug,
+      errorParams,
+      buildSteps,
+    })
+
+    const { success, severityCode } = getSeverity('success')
+
+    return { success, severityCode, netlifyConfig: netlifyConfigA, configMutations, logs }
+  } catch (error) {
+    const { severity } = await handleBuildError(error, errorParams)
+
+    const { success, severityCode } = getSeverity(severity)
+
+    return { success, severityCode, logs }
+  }
+}
+
+const startBuild = function (flags) {
+  const logs = getBufferLogs(flags)
+
+  logBuildStart(logs)
+
+  const { bugsnagKey, ...flagsA } = normalizeFlags(flags, logs)
+
+  const errorMonitor = startErrorMonitor({ flags: flagsA, logs, bugsnagKey })
+
+  return { ...flagsA, errorMonitor, logs }
+}
+
+const getBuildSteps = function (buildSteps) {
+  const allSteps = []
+
+  if (buildSteps.includes('edgeFunctionsBundling')) {
```

Comment:
Every core step exports an object that contains, alongside a few other properties, a `coreStepId`. In the case of the edge functions bundling step, this is `edge_functions_bundling`.

Rather than introducing a new nomenclature, manually checking what's in `buildSteps` and then doing a mapping for the actual core steps, I think we could use the values of `coreStepId` in `buildSteps` and do that mapping automatically.

Something like:

```js
const steps = [bundleEdgeFunctions].filter(({ coreStepId }) => buildSteps.includes(coreStepId))

return steps
```